Bluewire.Tools.Runner/FindBuild/TargetBranchResolver.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Threading.Tasks;
+using Bluewire.Common.GitWrapper;
+using Bluewire.Common.GitWrapper.Model;
+using Bluewire.Conventions;
+
+namespace Bluewire.Tools.Runner.FindBuild
+{
+    public class TargetBranchResolver
+    {
+        private readonly GitSession gitSession;
+        private readonly Common.GitWrapper.GitRepository repository;
+
+        public TargetBranchResolver(GitSession gitSession, Common.GitWrapper.GitRepository repository)
+        {
+            if (gitSession == null) throw new ArgumentNullException(nameof(gitSession));
+            if (repository == null) throw new ArgumentNullException(nameof(repository));
+            this.gitSession = gitSession;
+            this.repository = repository;
+        }
+
+        public async Task<StructuredBranch[]> IdentifyTargetBranchesOfCommit(Ref hash)
+        {
+            var inspector = new RepositoryStructureInspector(gitSession);

```

Comment:
It's a service object. It doesn't store any state and merely encapsulates logic. There's no particular need to keep it around between calls.